sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/util/RLimit.java

```diff
@@ -0,0 +1,87 @@
+package com.mesosphere.sdk.specification.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.mesos.Protos;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Representation of an individual rlimit, consisting of a name and optional soft/hard limits.
+ *
+ * A valid instance of this class has either both limits set or neither, with the further constraint that the soft limit
+ * must be less than or equal to the hard limit.
+ */
+public class RLimit {
+    private static final Set<String> RLIMITS = new HashSet<>();
```

Comment:
Could do the following to consolidate the enum<->String handling that's currently spread across here and `DefaultOfferRequirementProvider`:
- `DefaultOfferRequirementProvider`'s `RLIMIT_TYPE_MAP` could be moved here, replacing this `RLIMITS` Set
- `DefaultOfferRequirementProvider.getRLimitInfo()` can then call a `@JsonIgnore` `getEnum()` function which returns the `RLimit.Type`

(I was about to say 'what if the name isn't present in RLIMIT_TYPE_MAP?' in DefaultOfferRequirementProvider, and then realized that that check was effectively happening in `validate()` over here)